src/ar_config.hrl

```diff
@@ -0,0 +1,29 @@
+%% Start options with default values.
+-record(opts, {
```

Comment:
Seems like the term `config` is very well established by this branch. Wouldn't it make sense to rename this record to something with `config`? Maybe just `config`.